activesupport/lib/active_support/number_helper/number_to_currency_converter.rb

```diff
@@ -12,8 +12,11 @@ def convert
         format = options[:format]
 
         if number.to_f.negative?
-          format = options[:negative_format]
           number = absolute_value(number)
+
+          unless options[:precision] == 0 && number.to_f < 0.5
+            format = options[:negative_format]
+          end
         end
```

Comment:
`if number.to_f.negative?
   number = absolute_value(number) 
   format = options[:negative_format] unless options[:precision] == 0 && number.to_f < 0.5
end`
we can use like this also

Reply:
The negative_format option validation needs to be inside that if and using the unless condition in one line would be the same as it currently is. It can be changed to a one-line condition, but I'm not sure what are the current coding conventions on that.